src/backend/core/models.py

```diff
@@ -665,6 +631,87 @@ def get_abilities(self, user):
             MailboxAbilities.CAN_MANAGE_LABELS: can_modify,
         }
 
+    def get_validated_signature(self, signature_id: str, user: User):
+        """Helper method to validate and retrieve a signature template.
+
+        Args:
+            signature_id: ID of the signature template
+            user: User making the request
+
+        Returns:
+            MessageTemplate if valid and accessible, None otherwise
+        """
+        # Check for forced signature with domain having priority over mailbox
+        forced_signature = (
+            MessageTemplate.objects.filter(
+                Q(maildomain=self.domain) | Q(mailbox=self),
+                type=MessageTemplateTypeChoices.SIGNATURE,
+                is_forced=True,
+                is_active=True,
+            )
+            .order_by(
+                # Domain signatures first (maildomain_id not null), then mailbox signatures
+                Case(
+                    When(maildomain__isnull=False, then=0),
+                    default=1,
+                )
+            )
+            .first()
+        )
+
+        signature = forced_signature if forced_signature else None
+        if not signature and not signature_id:
+            return None
+
+        if not signature and signature_id:
+            try:
+                signature = MessageTemplate.objects.get(
+                    id=signature_id,
+                    type=MessageTemplateTypeChoices.SIGNATURE,
+                    is_active=True,
+                )
+            except MessageTemplate.DoesNotExist:
+                logger.error("Signature template not found with id: %s", signature_id)
+                return None
+
+            # Verify signature is in sender scope
+            in_sender_scope = (
+                signature.mailbox_id and signature.mailbox_id == self.id
+            ) or (signature.maildomain_id and signature.maildomain_id == self.domain_id)
+            if not in_sender_scope:
+                logger.warning(
+                    "User %s attempted to use signature %s outside sender scope",
+                    user.id,
+                    signature_id,
+                )
+                return None
+
+            # Verify user has access to the signature template
+            user_has_access = False
```

Comment:
@sdemagny I still don't understand why we need to pass `user` and check this here? If we remove all these checks the tests still pass.